### PR TITLE
Psionics firing pin

### PR DESCRIPTION
**Pull request by DreamySkrell**
lets the user fire the gun only if the user is capable of psionics (skrell)
does not work with the psionic implant (from the loadout)
should work with humans/etc with the borer stuff (as it gives psionics as well)

### CODE REVIEW

**Comment by DreamySkrell** [comment]
!review

**Review by RyverStyx — APPROVED** [review]

**Review by listerla — APPROVED** [review]

**Comment by Arrow768** [comment]
DNM until the (possibly unintentional) side effects of that are acknowledged and accepted by the skrell lore team.

File: code/modules/projectiles/pins.dm [hunk 1]
```diff
@@ -105,6 +105,18 @@ Pins Below.
...
 		return 0
 
+// Psionics pin, checks for psionics (psi aug not included)
+/obj/item/device/firing_pin/psionic
+	name = "psionics firing pin"
+	desc = "This is a psionics-locked firing pin which only authorizes users who are capable of psionics."
+	fail_message = "<span class='warning'>PSIONICS CHECK FAILED.</span>"
+
+/obj/item/device/firing_pin/psionic/pin_auth(mob/living/user)
+	if(user.can_commune() || user.psi)
```
Comment (NonQueueingMatt) [inline code/modules/projectiles/pins.dm:115]:
```suggestion
	if(user.can_commune())
```
`can_commune` already checks for the presence of psi.

**Review by NonQueueingMatt — CHANGES_REQUESTED** [review]

**Review by Dreamykass — APPROVED** [review]

**Comment by Arrow768** [comment]
The side-effects have been acknowledged by the skrell lore team via discord.

**Review by Arrow768 — APPROVED** [review]

**Comment by DreamySkrell** [comment]
This could be merged then.

**Review by NonQueueingMatt — APPROVED** [review]

**Pull request merged**